### PR TITLE
Add a callback on the reader options to expose the log filter switches

### PR DESCRIPTION
**Pull request by 0xced**
Just like it was done for log level switches in #352.

### CODE REVIEW

**Review by nblumhardt — APPROVED** [review]
LGTM :+1:

**Review by skomis-mm — COMMENTED** [review]
Looks good overall 👍  Just a minor concern around $ prefixing on callback calling

File: src/Serilog.Settings.Configuration/Settings/Configuration/ConfigurationReader.cs [hunk 1]
```diff
@@ -75,7 +75,8 @@ void ProcessFilterSwitchDeclarations()
             SetFilterSwitch(throwOnError: true);
             SubscribeToFilterExpressionChanges();
 
-            _resolutionContext.AddFilterSwitch(switchName, filterSwitch);
+            var referenceName = _resolutionContext.AddFilterSwitch(switchName, filterSwitch);
+            _resolutionContext.ReaderOptions.OnFilterSwitchCreated?.Invoke(referenceName, filterSwitch);
```
[the remark this answers is not in the document]
Comment (nblumhardt) [reply src/Serilog.Settings.Configuration/Settings/Configuration/ConfigurationReader.cs:79]: @0xced any thoughts on this one? (I'm working on merging everything through for a v7 shortly, would be great to get this PR in however this goes)

[the remark this answers is not in the document]
Comment (0xced) [reply src/Serilog.Settings.Configuration/Settings/Configuration/ConfigurationReader.cs:79]: I haven't forgotten about this, I will reply soon but I want to experiment with a real world scenario to find out what's best first.

[the remark this answers is not in the document]
Comment (0xced) [reply src/Serilog.Settings.Configuration/Settings/Configuration/ConfigurationReader.cs]: I agree with you that there's no need to force the leading `$`. So I addressed this in f8813e974b107fa48df8d3aa4cc695c68d0fce5c.

**Pull request merged**